### PR TITLE
Dockerize #181

### PR DESCRIPTION
**Pull request by cricket007**
# stream-registry PR

Same commits as #181 that was (minimally) Dockerized

One commit -  4a7670b

### Added
* Docker maven plugins w/ profile 
* Dockerfile to `app` module; Dependency plugin to copy over api graphql file, otherwise got errors about file not on the app's classpath. 

### Changed
* `maven-jar-plugin` to add classpath and main-class entries in `app`

# To build

For internal usage

`./mvnw clean package -DskipTests -Ddocker.repo=<docker.repo>/<owner>/` (trailing slash is important) && `docker push <docker.repo>/<owner>/stream-registry` (until the plugin is setup to push)

by default, just builds local image `stream-registry` 

# PR Checklist Forms

- [ ] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: app/pom.xml [hunk 1]
```diff
@@ -40,4 +40,32 @@
...
+          <execution>
+            <id>default</id>
+            <goals>
+              <goal>build</goal>
+              <goal>push</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <repository>dl2.homeawaycorp.com/expediagroup/${project.artifactId}</repository>
```
Comment (cricket007) [inline app/pom.xml]: TODO: Change the repository

File: app/pom.xml [hunk 2]
```diff
@@ -40,4 +40,32 @@
...
+          <tag>${project.version}</tag>
+          <buildArgs>
+            <CONTEXT_DIR>src/main/docker</CONTEXT_DIR>
+            <JAR_FILE>${project.build.finalName}.jar</JAR_FILE>
+            <!--<JAR_DEPENDENCIES>target/lib</JAR_DEPENDENCIES>-->
```
Comment (cricket007) [inline app/pom.xml]: Can remove this

File: app/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM openjdk:11
+
+ENTRYPOINT ["/docker-entrypoint.sh"]
+
+ARG CONTEXT_DIR
```
Comment (cricket007) [inline app/Dockerfile]: Would be nice if we could add in some `application.yaml`

File: app/src/main/resources/stream-registry.graphql [hunk 4]
```diff
@@ -0,0 +1,102 @@
+type KeyValue {
```
Comment (cricket007) [inline app/src/main/resources/stream-registry.graphql]: I had to copy this file from another module. 

Try to build & run the container without this here, and it says cannot find `**/*.graphql` files on the classpath.

File: app/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+ADD target/lib /usr/share/stream-registry/lib
```
Comment (cricket007) [inline app/Dockerfile]: Docker doesn't require uber-jars

File: pom.xml [hunk 6]
```diff
@@ -1,866 +1,543 @@
...
+            <artifactId>dockerfile-maven-plugin</artifactId>
+            <version>1.4.10</version>
+            <executions>
+              <execution>
+                <id>default</id>
+                <goals>
+                  <goal>build</goal>
+                </goals>
+                <configuration>
+                  <repository>${docker.repo}${docker.image.name}</repository>
```
Comment (cricket007) [inline pom.xml]: Could `expediadotcom` be added here?

File: pom.xml [hunk 7]
```diff
@@ -1,866 +1,545 @@
...
+      <id>docker</id>
+      <activation>
+        <file>
+          <exists>Dockerfile</exists>
+        </file>
+      </activation>
+      <properties>
+        <docker.repo/>
+        <docker.image.name>stream-registry</docker.image.name>
+        <docker.image.base>openjdk:11</docker.image.base>
```
Comment (cricket007) [inline pom.xml]: Would like to use existing, internal Docker image cache. 

Needs documented somewhere how to change the base image, though. E.g.

`mvn clean package -Ddocker.image.base=docker.registry.internal/java:11`

**Comment by neoword** [comment]
@cricket007 can you resolve conflicts on this ?

**Comment by neoword** [comment]
Please close and resubmit to `vrbo-master`

**Comment by cricket007** [comment]
> resubmit to vrbo-master and I'll happily approve.

Can't do that because the changes depend on the GraphQL implementation in the `eg-master` branch. `vrbo-master` is still using the `jib-maven-plugin`.

**Comment by cricket007** [comment]
Why was this closed?

**Comment by neoword** [comment]
@cricket007 can you re-open a new PR against `vrbo-master` please?

**Comment by cricket007** [comment]
It applies to `eg-master`, as mentioned.

**Comment by nahguam** [comment]
Apologies, I think this got closed because I deleted `eg-master` now that it's moved to `master`. 

Could we consider using jib as it does away with most of the boilerplate?

**Comment by neoword** [comment]
Ahh... I thought this was for `vrbo-master` not `eg-master`. I stand corrected.

**Comment by cricket007** [comment]
In any case, #189 uses Jib, as requested

**Pull request closed without merging**